Source/WebCore/css/parser/CSSPropertyParserConsumer+Image.cpp

```diff
@@ -0,0 +1,1185 @@
+{
+    // https://compat.spec.whatwg.org/#css-gradients-webkit-radial-gradient/ states that -webkit-radial-gradient() and
+    // -webkit-repeating-radial-gradient() must be "treated as an alias of radial-gradient as defined in [css3-images-20110217]."
+    // In [css3-images-20110217] the grammar was defined as:
+    //
+    //   <radial-gradient> = radial-gradient([<bg-position>,]? [ [<shape> || <size>] | [<length> | <percentage>]{2} ,]? <color-stop>[, <color-stop>]+);
+    //   <shape> = [ circle | ellipse ]
+    //   <size> = [ closest-side | closest-corner | farthest-side | farthest-corner | contain | cover ]
+    //
+    //      defaults to ‘ellipse cover’.
+    //
+    // see https://www.w3.org/TR/2011/WD-css3-images-20110217/#radial-gradients.
+
+    static constexpr std::pair<CSSValueID, CSSPrefixedRadialGradientValue::ShapeKeyword> shapeMappings[] {
+        { CSSValueCircle, CSSPrefixedRadialGradientValue::ShapeKeyword::Circle },
+        { CSSValueEllipse, CSSPrefixedRadialGradientValue::ShapeKeyword::Ellipse },
+    };
+    static constexpr SortedArrayMap shapeMap { shapeMappings };
+
+    static constexpr std::pair<CSSValueID, CSSPrefixedRadialGradientValue::ExtentKeyword> extentMappings[] {
+        { CSSValueContain, CSSPrefixedRadialGradientValue::ExtentKeyword::Contain },
+        { CSSValueCover, CSSPrefixedRadialGradientValue::ExtentKeyword::Cover },
+        { CSSValueClosestSide, CSSPrefixedRadialGradientValue::ExtentKeyword::ClosestSide },
+        { CSSValueClosestCorner, CSSPrefixedRadialGradientValue::ExtentKeyword::ClosestCorner },
+        { CSSValueFarthestSide, CSSPrefixedRadialGradientValue::ExtentKeyword::FarthestSide },
+        { CSSValueFarthestCorner, CSSPrefixedRadialGradientValue::ExtentKeyword::FarthestCorner },
+    };
+    static constexpr SortedArrayMap extentMap { extentMappings };
+
+    std::optional<CSSGradientPosition> position;
+    if (auto centerCoordinate = consumeOneOrTwoValuedPositionCoordinates(range, context.mode, UnitlessQuirk::Forbid)) {
+        if (!consumeCommaIncludingWhitespace(range))
+            return nullptr;
+        position = CSSGradientPosition { WTFMove(centerCoordinate->x), WTFMove(centerCoordinate->y) };
+    }
+
+    std::optional<CSSPrefixedRadialGradientValue::ShapeKeyword> shapeKeyword;
+    std::optional<CSSPrefixedRadialGradientValue::ExtentKeyword> extentKeyword;
+    if (range.peek().type() == IdentToken) {
+        shapeKeyword = consumeIdentUsingMapping(range, shapeMap);
+        extentKeyword = consumeIdentUsingMapping(range, extentMap);
+        if (!shapeKeyword)
+            shapeKeyword = consumeIdentUsingMapping(range, shapeMap);
+        if (shapeKeyword || extentKeyword) {
+            if (!consumeCommaIncludingWhitespace(range))
+                return nullptr;
+        }
+    }
+
+    CSSPrefixedRadialGradientValue::GradientBox gradientBox = std::monostate { };
```

Comment:
Is `= std::monostate { }` needed? Why isn’t that what the default constructor makes?

Reply:
I prefer the explicit construction since the default value of a variant is based on position and an innocuous looking change to the order in the definition would potentially change behavior.